dataland-frontend/tests/e2e/specs/framework-view-and-upload-journey/EditDataset.ts

```diff
@@ -0,0 +1,101 @@
+import { getKeycloakToken } from "@e2e/utils/Auth";
+import { reader_name, reader_pw, uploader_name, uploader_pw } from "@e2e/utils/Cypress";
+import {
+  CompanyDataControllerApi,
+  Configuration,
+  DataTypeEnum,
+  LksgData,
+  LksgDataControllerApi,
+} from "@clients/backend";
+import { FixtureData } from "@e2e/fixtures/FixtureUtils";
+import { getPreparedFixture } from "@e2e/utils/GeneralApiUtils";
+import { uploadCompanyAndLksgDataViaApi } from "@e2e/utils/LksgUpload";
+import { describeIf } from "../../support/TestUtility"; // TODO use @ in path
+
+describeIf(
+  "Validates the edit button functionality on the view framework page",
+  {
+    executionEnvironments: ["developmentLocal", "ci"],
+    dataEnvironments: ["fakeFixtures", "realData"],
+  },
+  () => {
+    /**
+     * Uses the dataland API to retrieve the id of an arbitrary company with data for the specified data framework
+     *
+     * @param dataType the framework to search a company for
+     * @returns a cypress chainable containing the id of a company with data for the specified framework
+     */
+    function getFirstCompanyIdFromApiWithFramework(dataType: DataTypeEnum): Cypress.Chainable<string> {
+      return getKeycloakToken(reader_name, reader_pw).then(async (token) => {
+        const data = await new CompanyDataControllerApi(new Configuration({ accessToken: token })).getCompanies(
+          undefined,
+          new Set([dataType])
+        );
+        return data.data[0].companyId;
+      });
+    }
+
+    it("Should not display the edit and create new dataset button on the framework view page for a data reader", () => {
+      getFirstCompanyIdFromApiWithFramework(DataTypeEnum.Lksg).then((companyId) => {
+        cy.ensureLoggedIn(reader_name, reader_pw);
+        cy.visit(`/companies/${companyId}/frameworks/lksg`);
+        cy.get("h2").contains("LkSG Data").should("be.visible");
+        cy.get("button:contains(EDIT)").should("not.exist");
+        cy.get("button:contains(NEW DATASET)").should("not.exist");
+      });
+    });
+
+    it("Should only display the add new dataset button for a framework for which the add new dataset functionality is not implemented", () => {
+      getFirstCompanyIdFromApiWithFramework(DataTypeEnum.Sfdr).then((companyId) => {
+        cy.ensureLoggedIn(uploader_name, uploader_pw);
+        cy.visit(`/companies/${companyId}/frameworks/sfdr`);
+        cy.get("h2").contains("SFDR data").should("be.visible");
+        cy.get("button:contains(EDIT)").should("not.exist");
+        cy.get("button:contains(NEW DATASET)").should("exist").click();
```

Comment:
If here we only check if the correct link is contained in the button the whole test is a component test again.

Reply:
Migrated to Component Test in "ViewFrameworkBase.cy.ts"